webapp/src/features/adminAdduser/AdminAddUser.js

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect } from "react";
+import { Form } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import { Multiselect, DropdownList } from "react-widgets";
+import {
+  selectSuccessMsg,
+  selectSelectedOrganizationNames,
+  selectSelectedOrganizations,
+  selectOrganizationNames,
+  selectAvailableRoles,
+  selectApiData,
+  selectErrorState,
+  selectErrorMsg,
+  selectSubmitAttempt,
+
+  // actions
+  getUserData,
+  setSelectedRole,
+  updateOrganizationNamesApiData,
+  updateAvailableRolesApiData,
+  updateOrganizations,
+  submitForm,
+} from "./adminAddUserSlice";
+import { useSelector, useDispatch } from "react-redux";
+
+import "../adminRsuTab/Admin.css";
+import "react-widgets/styles.css";
+
+const AdminAddUser = (props) => {
+  const dispatch = useDispatch();
+  const successMsg = useSelector(selectSuccessMsg);
+  const selectedOrganizationNames = useSelector(selectSelectedOrganizationNames);
+  const selectedOrganizations = useSelector(selectSelectedOrganizations);
+  const organizationNames = useSelector(selectOrganizationNames);
+  const availableRoles = useSelector(selectAvailableRoles);
+  const apiData = useSelector(selectApiData);
+  const errorState = useSelector(selectErrorState);
+  const errorMsg = useSelector(selectErrorMsg);
+  const submitAttempt = useSelector(selectSubmitAttempt);
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm();
+
+  useEffect(() => {
+    dispatch(getUserData());
+  }, []);
```

Comment:
Should we include dispatch in the dependency array?

Reply:
added